### PR TITLE
Create proper README

### PR DESCRIPTION
**Pull request by normandy7**
Create REAMDE with setup instructions and API reference.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,373 @@
-### Local Environment
-Create new virtual environment and run `pip install -r dev_requirements.txt`
-Run `pre-commit install` to initialize [pre-commit](https://pre-commit.com/)
+# Neptune Scale client
+
+> [!NOTE]
+> This package only works with the `3.0` version of neptune.ai called Neptune Scale, which is in beta.
+>
+> It's supported on Linux and MacOS.
```
Comment (Raalsky) [inline README.md:6]: All as mentioned here: https://github.com/neptune-ai/neptune-client-scale/pull/15#discussion_r1742182055

File: README.md [hunk 2]
```diff
@@ -1,3 +1,373 @@
...
+
+> If you're a workspace admin, you can also set up a service account. This way, multiple people or machines can share the same API token.
+
+Install the neptune-scale package:
+
+```
+pip install neptune-scale
+```
+
+In the environment where neptune-scale is installed, set the following environment variables to the API token and project name obtained from the Neptune beta instance:
```
Comment (Raalsky) [inline README.md]: Maybe let's do not use "beta" phrasing. Just "Scale" is enough.

File: README.md [hunk 3]
```diff
@@ -1,3 +1,373 @@
...
+)
+
+run.log_configs(
+    data={"learning_rate": 0.001, "batch_size": 64}
+)
+
+# inside a training loop
+for e in epoch:
+    run.log_metrics(
+        data={"acc": 0.89, "loss": 0.17},
```
Comment (Raalsky) [inline README.md:63]: ⚠️ We have to provide `step` here (and increasing) as it won't work properly.

File: README.md [hunk 4]
```diff
@@ -1,3 +1,373 @@
...
+| `run_id`         | `str`            | -       | Identifier of the run. Must be unique within the project. Max length: 128 bytes. |
+| `project`        | `str`, optional  | `None`  | Name of a project in the form `workspace-name/project-name`. If `None`, the value of the `NEPTUNE_PROJECT` environment variable is used. |
+| `api_token`      | `str`, optional  | `None`  | Your Neptune API token or a service account's API token. If `None`, the value of the `NEPTUNE_API_TOKEN` environment variable is used. To keep your token secure, don't place it in source code. Instead, save it as an environment variable. |
+| `resume`         | `bool`, optional | `False` | If `False` (default), creates a new run. To continue an existing run, set to `True` and pass the ID of an existing run to the `run_id` argument. To fork a run, use `fork_run_id` and `fork_step` instead. |
+| `mode`           | `Literal`, `"async"` or `"disabled"` | `"async"` | Mode of operation. If set to `"disabled"`, the run doesn't log any metadata. |
```
Comment (Raalsky) [inline README.md:105]: No need to mention the `Literal` that's rather typing specific and we can use `enum` instead.

File: README.md [hunk 5]
```diff
@@ -1,3 +1,373 @@
...
+| `experiment_name`  | `str`, optional  | `None` | Name of the experiment to associate the run with. Learn more about [experiments](https://docs-beta.neptune.ai/experiments) in the Neptune documentation. |
+| `creation_time`  | `datetime`, optional | `None` | Custom creation time of the run. |
+| `fork_run_id`    | `str`, optional  | `None` | If forking off an existing run, ID of the run to fork from. |
```
Comment (Raalsky) [inline README.md:108]: I think that `ID of the run to fork from` should be enough. You always have to fork from an existing run.

Comment (Raalsky) [reply README.md:108]:
```suggestion
| `fork_run_id`    | `str`, optional  | `None` | The ID of the run to fork from. |
```

File: README.md [hunk 6]
```diff
@@ -1,3 +1,373 @@
...
+| `fork_step`      | `int`, optional  | `None` | If forking off an existing run, step number to fork from. |
```
Comment (Raalsky) [inline README.md:109]:
```suggestion
| `fork_step`      | `int`, optional  | `None` | The step number to fork from. |
```

File: README.md [hunk 7]
```diff
@@ -1,3 +1,373 @@
...
+- `value`: the piece of metadata to log.
+
+For example, `{"parameters/learning_rate": 0.001}`. In the field path, each forward slash `/` nests the field under a namespace. Use namespaces to structure the metadata into meaningful categories.
+
+__Parameters__
+
+| Name          | Type                                               | Default | Description                                                               |
+|---------------|----------------------------------------------------|---------|---------------------------------------------------------------------------|
+| `timestamp`   | `datetime`, optional                               | `None`  | Time of logging the metadata. |
+| `data`      | `Dict[str, Union[float, bool, int, str, datetime, list, set]]`, optional  | `None` | Dictionary of configs or other values to log. Available types: float, integer, Boolean, string, and datetime. |
```
Comment (Raalsky) [inline README.md:204]: We should either remove the `list` and `set` or mention that StringSets are available:
```python
run.log_configs(
        data={"learning_rate": 0.001, "batch_size": 64, "test": ["1", "2", "3"]}
    )
```

File: README.md [hunk 8]
```diff
@@ -1,3 +1,373 @@
...
+__Examples__
+
+Create a run and log metrics:
+
+```python
+from neptune_scale import Run
+
+with Run(...) as run:
+    run.log_metrics(
+        data={"loss": 0.14, "acc": 0.78},
```
Comment (Raalsky) [inline README.md:250]: Let's provide the `step` value as well as it almost doesn't make sense to use this method without it.

File: README.md [hunk 9]
```diff
@@ -1,3 +1,369 @@
...
+| `project`        | `str`, optional  | `None`  | Name of a project in the form `workspace-name/project-name`. If `None`, the value of the `NEPTUNE_PROJECT` environment variable is used. |
+| `api_token`      | `str`, optional  | `None`  | Your Neptune API token or a service account's API token. If `None`, the value of the `NEPTUNE_API_TOKEN` environment variable is used. To keep your token secure, don't place it in source code. Instead, save it as an environment variable. |
+| `resume`         | `bool`, optional | `False` | If `False` (default), creates a new run. To continue an existing run, set to `True` and pass the ID of an existing run to the `run_id` argument. To fork a run, use `fork_run_id` and `fork_step` instead. |
+| `mode`           | `"async"` or `"disabled"` | `"async"` | Mode of operation. If set to `"disabled"`, the run doesn't log any metadata. |
+| `experiment_name`  | `str`, optional  | `None` | Name of the experiment to associate the run with. Learn more about [experiments](https://docs-beta.neptune.ai/experiments) in the Neptune documentation. Max length: 730 bytes. |
```
Comment (Raalsky) [inline README.md:105]: I don't think it's worth to mention it. We've removed it in https://github.com/neptune-ai/neptune-client-scale/pull/25#discussion_r1741747923

Comment (Raalsky) [reply README.md:105]:
```suggestion
| `experiment_name`  | `str`, optional  | `None` | Name of the experiment to associate the run with. Learn more about [experiments](https://docs-beta.neptune.ai/experiments) in the Neptune documentation. |
```

**Review by Raalsky — APPROVED** [review]
